python/cudf/cudf/core/udf/strings_lowering.py

```diff
@@ -0,0 +1,302 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+import operator
+
+from numba import types
+from numba.core import cgutils
+from numba.cuda.cudaimpl import lower as cuda_lower
+
+from strings_udf._typing import string_view
+from strings_udf.lowering import (
+    string_view_contains_impl,
+    string_view_endswith_impl,
+    string_view_find_impl,
+    string_view_isalnum_impl,
+    string_view_isalpha_impl,
+    string_view_isdecimal_impl,
+    string_view_isdigit_impl,
+    string_view_islower_impl,
+    string_view_isspace_impl,
+    string_view_isupper_impl,
+    string_view_len_impl,
+    string_view_rfind_impl,
+    string_view_startswith_impl,
+)
+
+from cudf.core.udf.masked_typing import MaskedType
+
+
+@cuda_lower(len, MaskedType(string_view))
+def masked_string_view_len_impl(context, builder, sig, args):
+    ret = cgutils.create_struct_proxy(sig.return_type)(context, builder)
+    masked_sv_ty = sig.args[0]
+    masked_sv = cgutils.create_struct_proxy(masked_sv_ty)(
+        context, builder, value=args[0]
+    )
+    result = string_view_len_impl(
+        context, builder, types.int32(string_view), (masked_sv.value,)
+    )
+    ret.value = result
+    ret.valid = masked_sv.valid
+
+    return ret._getvalue()
+
+
+@cuda_lower(
+    "MaskedType.startswith", MaskedType(string_view), MaskedType(string_view)
+)
+def masked_string_view_startswith_impl(context, builder, sig, args):
```

Comment:
Can we call another function inside of this that dispatches on `string_view_startswith_impl` so that we don't have allllll this boilerplate in every function? Maybe some boilerplate is shared by functions returning boolean types, other boilerplate returned by integer (size) types?